### PR TITLE
Ash script outputs total number of stations connected to all network …

### PR DESCRIPTION
**Pull request by jasonkhanlar**
…devices of a node

### CODE REVIEW

File: mm-count-associated-devices.ash [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+count=0
+iw dev | \
+grep Interface | \
+{
+  while IFS= read -r line;do
+    dev="$(echo $line|cut -d " " -f 2)"
+    stations="$(iw dev $dev station dump|grep Station|wc -l)"
+    count=$(expr $count + $stations)
+  done
+  cat $count
```
Comment (colindean) [inline mm-count-associated-devices.ash]: `cat`? or `echo`?

File: mm-count-associated-devices.ash [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/ash
```
Comment (colindean) [inline mm-count-associated-devices.ash]: Any particular reason this is bound to `ash` and not `sh`?

**Comment by jasonkhanlar** [comment]
Rather than requiring each node to have this script to calculate and output how many nodes are connected to it, it should be possible to obtain this information without any scripts using SNMP to communicate with the device.

For example, 1 test node has 2 interfaces (wlan0 and wlan0-1) which show a total of 12 connected devices:

```
root@rocketm2-workhardpgh-2401:~# iw dev wlan0 station dump|grep Station
Station 68:72:51:38:03:b3 (on wlan0)
Station c4:6e:1f:88:e5:6c (on wlan0)
Station 24:a4:3c:40:d1:be (on wlan0)
Station 24:a4:3c:9a:11:49 (on wlan0)
Station 24:a4:3c:7e:73:7c (on wlan0)
Station 24:a4:3c:70:a4:08 (on wlan0)
Station 24:a4:3c:7e:73:4b (on wlan0)
Station 68:72:51:22:31:52 (on wlan0)
Station 24:a4:3c:fe:05:51 (on wlan0)
root@rocketm2-workhardpgh-2401:~# iw dev wlan0-1 station dump|grep Station
Station 74:a5:28:16:62:1b (on wlan0-1)
Station 7c:0b:c6:5c:2e:a4 (on wlan0-1)
Station 80:d2:1d:2d:fa:dc (on wlan0-1)
```

Attempting to get total number of devices connected to a node using SNMP, retrieving those same MAC addresses, the following command outputs 25 devices (13 too many):

```
jason@localcomputer # snmpwalk -v1 -c HardwareSNMP rocketm2-workhardpgh-2401|grep IP-MIB::ipNetToMediaPhysAddress
IP-MIB::ipNetToMediaPhysAddress.2.100.65.120.168 = STRING: 1c:c1:de:1b:78:a8
IP-MIB::ipNetToMediaPhysAddress.2.100.92.10.220 = STRING: 1c:c1:de:1c:a:de
IP-MIB::ipNetToMediaPhysAddress.2.172.16.32.1 = STRING: 1c:c1:de:1b:78:a8
IP-MIB::ipNetToMediaPhysAddress.2.172.16.32.201 = STRING: 24:a4:3c:79:6e:3c
IP-MIB::ipNetToMediaPhysAddress.2.172.16.33.1 = STRING: 24:a4:3c:9b:11:49
IP-MIB::ipNetToMediaPhysAddress.2.172.16.34.11 = STRING: 0:30:48:c7:16:58
IP-MIB::ipNetToMediaPhysAddress.2.172.16.34.181 = STRING: d6:23:2a:86:ce:55
IP-MIB::ipNetToMediaPhysAddress.2.172.16.35.43 = STRING: 2c:33:7a:80:c3:1f
IP-MIB::ipNetToMediaPhysAddress.2.172.16.35.58 = STRING: 5a:c6:91:62:df:eb
IP-MIB::ipNetToMediaPhysAddress.2.172.16.35.98 = STRING: e4:95:6e:40:74:85
IP-MIB::ipNetToMediaPhysAddress.5.100.65.209.190 = STRING: 24:a4:3c:40:d1:be
IP-MIB::ipNetToMediaPhysAddress.5.100.73.229.107 = STRING: c4:6e:1f:88:e5:6c
IP-MIB::ipNetToMediaPhysAddress.5.100.91.17.73 = STRING: 24:a4:3c:9a:11:49
IP-MIB::ipNetToMediaPhysAddress.5.100.99.49.82 = STRING: 68:72:51:22:31:52
IP-MIB::ipNetToMediaPhysAddress.5.100.113.164.8 = STRING: 24:a4:3c:70:a4:8
IP-MIB::ipNetToMediaPhysAddress.5.100.121.3.179 = STRING: 68:72:51:38:3:b3
IP-MIB::ipNetToMediaPhysAddress.5.100.127.5.81 = STRING: 24:a4:3c:fe:5:51
IP-MIB::ipNetToMediaPhysAddress.5.100.127.115.75 = STRING: 24:a4:3c:7e:73:4b
IP-MIB::ipNetToMediaPhysAddress.5.100.127.115.124 = STRING: 24:a4:3c:7e:73:7c
IP-MIB::ipNetToMediaPhysAddress.6.10.92.248.42 = STRING: 88:7:4b:d6:27:f1
IP-MIB::ipNetToMediaPhysAddress.6.10.92.248.43 = STRING: 80:d2:1d:2d:fa:dc
IP-MIB::ipNetToMediaPhysAddress.6.10.92.248.74 = STRING: 74:a5:28:16:62:1b
IP-MIB::ipNetToMediaPhysAddress.6.10.92.248.178 = STRING: 7c:b:c6:5c:2e:a4
IP-MIB::ipNetToMediaPhysAddress.6.10.92.248.179 = STRING: bc:6c:21:b1:7a:d7
IP-MIB::ipNetToMediaPhysAddress.6.169.254.145.155 = STRING: c0:38:96:82:36:3
```

There are 9 IP-MIB::ipNetToMediaPhysAddress.5.\* results
and 9 results from # iw dev wlan0 station dump|grep Station
where physical address 5 == wlan0 is assumed

but assuming that physical address 6 == wlan0-1,
there are 6 IP-MIB::ipNetToMediaPhysAddress.6.\* results
and 3 results from # iw dev wlan0-1 station dump|grep Station
where 3 match, but with 3 extra mac address.
I do not understand why.

Does anyone have any ideas?

File: mm-count-associated-devices.ash (hunk 2 above)
Comment (jasonkhanlar) [inline mm-count-associated-devices.ash]: echo $SHELL outputs /bin/ash on some devices, and previous bash shell scripts were not compatible. I suppose .sh is sufficient, but I had initially used .ash. Should I change to .sh?

Comment (colindean) [inline mm-count-associated-devices.ash]: More concerned about the `#!/bin/ash` than the extension.

**Pull request merged**